File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        sort(strs.begin(), strs.end());
        int i = 0;
        string firstString = strs[0], lastString = strs.back();

        while(i < firstString.length() && firstString[i] == lastString[i]) i++;

        return lastString.substr(0, i);

    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] str) {
        if(str.length == 0 || str == null) return "";

        String prefix = str[0];
        

        for(int i = 1; i < str.length; i++){
            int j = 0;
            while (j < prefix.length() && j < str[i].length() && prefix.charAt(j) == str[i].charAt(j)){
                j++;
            }
            prefix = prefix.substring(0, j);
        }
        if(prefix.isEmpty()) return "";
        return prefix;

    }
}